### DISCUSSION TITLE
Please share your working Raspberry Pi HQ camera settings

### DISCUSSION THREAD

**Discussion by mikaelnousiainen** [aaronwmorris/indi-allsky#1261] [Q&A] [answered]
I've had great difficulties getting good *night-time* images with Raspberry Pi HQ camera. Daytime images are almost always just fine (in both JPG and RAW modes)

I've gone through the recommended settings for RAW image capture, and while the quality is very good in general (less noise and artifacts than with JPG mode with AWB enabled), I always seem to get weird colors (orange or purple) in low-light conditions. I couldn't get the software AWB work correctly.

With JPG mode + AWB enabled, there problems are: gaps between exposure (expected with AWB), weird noise in the images (not present in raw mode) and, worst of all, certain amount of light produced BLUE images (when the sky is not dark yet).

Could someone share working settings as a *starting point* for Raspberry Pi HQ? RAW mode would be preferred, but I can live with JPG mode if it would produce good colors consistently.

In JPG mode, the blue images appear as follows (two consecutive photos).

![ccd1_20240510_233431](https://github.com/aaronwmorris/indi-allsky/assets/3470400/90efea4f-da9d-445a-8fa8-3a8b3f3bee5f)
![ccd1_20240510_233446](https://github.com/aaronwmorris/indi-allsky/assets/3470400/d6a2b0c8-14f6-4992-83d0-eb3472cd8d1a)

**Comment by aaronwmorris** [comment]
Just as a starting point, do you know if your HQ camera has an IR-Cut filter or not?

I would try to avoid the 16-bit CLAHE algorithm, it has a bug in it. 

As a general recommendation, in RAW mode, I would enable SCNR - Average Neutral in RAW mode.  This can fix a lot of the color balance issues caused by Green bias.

**Reply by mikaelnousiainen** [reply]
It's a normal "stock" HQ camera, so I'd assume IR-Cut filter is there. Daytime photos have "normal" colors, which I think get altered if there is no IR-Cut filter.

**Reply by mikaelnousiainen** [reply]
I'm mainly using this for aurora / meteors, so natural colors would be more important for me.

**Reply by mikaelnousiainen** [reply]
Would you recommend using the software AWB?

**Reply by mikaelnousiainen** [reply]
For the record, with software AWB enabled (all colors at 1.0 correction), CLAHE disabled, SCNR in average netural mode I get very weird daytime colors, see:

![indi-allsky-raw-colors-mess](https://github.com/aaronwmorris/indi-allsky/assets/3470400/15ad9a00-1e9a-4676-a14b-0f96693285b7)

**Reply by aaronwmorris** [reply]
How old is this install of indi-allsky?  This looks like a problem with the Average Neutral SCNR I fixed a little over 6 months ago.  Can you update to the latest code?

https://github.com/aaronwmorris/indi-allsky/pull/954

**Reply by mikaelnousiainen** [reply]
Aha ok, the version I had was from last September. I've upgraded now, let's see what the results are once it gets brighter again tomorrow.

**Reply by mikaelnousiainen** [reply]
Ok, after the upgrade the bug with the colors is gone. However, now the overall tone is very much on red/orange side (see attached images). The same red tint was also present during night-time images. I'm still running the same settings with software AWB, SCNR Average mode and no CLAHE. No other correction is applied.

What would be your recommendation to reduce the red/yellow levels? Would the same change/values work for both night and day?

![indi-allsky-raw-colors-red](https://github.com/aaronwmorris/indi-allsky/assets/3470400/5765eeb8-6b21-416c-b980-3d2f4194e0c4)
![ccd1_20240514_010002](https://github.com/aaronwmorris/indi-allsky/assets/3470400/2f1df09f-a8a0-4c48-b1f5-3e02a7f87feb)

**Reply by caphector** [reply, marked as answer]
I spent a while tweaking my settings with the Pi HQ Camera. I ended up with:

Camera tab: 
 * Night Enable AWB: off
 * Day Enable AWB: off

Image tab:
 * Bayer Pattern: BGGR (likely does not need to be re-configured)
 * Red Balance Factor: 3.22
 * Green Balance Factor: 1
 * Blue Balance Factor: 1.5
 * Auto White Balance: Off
 * Saturation factor: 1.3
 * SCNR: Disabled

These settings are tested to the "Works for me™" standard. I do not use CLAHE, Daytime Contrast Enhance, Night time Contrast Enhance, or Stretching. Let me know if there are any other settings I can check.

**Reply by aaronwmorris** [reply]
Looks good!  Color balance is fairly difficult when dealing with RAW images.  With the libcamera cameras, you can turn on RAW/DNG mode and AWB mode and watch the log to see what color factors to set for Red and Blue to get a good starting point for your values.

**Reply by mikaelnousiainen** [reply]
Thanks @caphector ! I need to have slightly different balance factors (more green, less blue), though I did have SCNR average enabled. Need to test without that too! I think your config is a good starting point so I'll mark it as an answer.

**Comment by aaronwmorris** [comment]
What does it look like without the software AWB?

**Reply by mikaelnousiainen** [reply]
Thanks, disabling the software AWB fixes most of the colors. Now it's just a matter of fine-tuning the balance factors.
